packages/react/src/search/useSearch.ts

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { GenericFields, SearchParameters } from '@sitecore-content-sdk/search';
+import { DEFAULT_PAGE, DEFAULT_PAGE_SIZE, getOffset, useSearchService } from './utils';
+
+/**
+ * Options for the useSearch hook.
+ * @public
+ */
+export interface UseSearchOptions<T = GenericFields> {
+  /**
+   * The query string to search for.
+   * By default empty string is used.
+   */
+  query?: string;
+  /**
+   * The current page number (1-indexed).
+   * If not provided, the search will start on page 1.
+   */
+  page?: number;
+  /**
+   * The ID of the search index to use.
+   */
+  searchIndexId: string;
+  /**
+   * Number of results per page.
+   * @default 10
+   */
+  pageSize?: number;
+  /**
+   * Specifies the sorting of the search results.
+   */
+  sort?: SearchParameters<T>['sort'];
+}
+
+/**
+ * The state of the useSearch hook.
+ * @public
+ */
+export interface UseSearchState<T = GenericFields> {
+  /**
+   * The search results.
+   */
+  results: T[];
+  /**
+   * Whether a search request is currently in progress.
+   */
+  isLoading: boolean;
+  /**
+   * Whether the search request was successful.
+   */
+  isSuccess: boolean;
+  /**
+   * Whether the search request failed.
+   */
+  isError: boolean;
+  /**
+   * Total number of results across all pages.
+   */
+  total: number;
+  /**
+   * Total number of pages available based on `total` and `pageSize`.
+   */
+  totalPages: number;
+  /**
+   * The error object if the last search request failed, or null if no error occurred.
+   */
+  error: Error | null;
+}
+
+/**
+ * React hook for performing search queries with pagination.
+ * @param {UseSearchOptions} options - Configuration options for the search hook.
+ * @returns {UseSearchState} The search state.
+ * @public
+ */
+export const useSearch = <T = GenericFields>(
```

Comment:
Currently hook fires immediately and there is no way to call conditionally, e.g. in button click.
add `enabled` option that controls if search request is executed (default is true)